### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by rmtolmach**
Admins are now able to create new categories and edit existing ones. Added a conditional to check if the admin was creating or editing by seeing if there was an id in the params. And changed some flash notice logic.

I wrote a test that checks if the new category page loads.

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -25,12 +25,13 @@ def destroy
 
   def new_or_edit
     @categories = Category.find(:all)
-    @category = Category.find(params[:id])
+    @category = Category.new
```
Comment (kariabancroft) [inline app/controllers/admin/categories_controller.rb]: It seems like this is redundant, you should only create the new category if there is no params present

File: spec/controllers/admin/categories_controller_spec.rb [hunk 2]
```diff
@@ -11,6 +11,11 @@
     request.session = { :user => henri.id }
   end
 
+  it "test_new" do
+    get :new
+    response.should be_success
```
Comment (kariabancroft) [inline spec/controllers/admin/categories_controller_spec.rb:16]: This test may also want to test the template that is rendered

Comment (kariabancroft) [inline spec/controllers/admin/categories_controller_spec.rb]: This test may also want to test the template that is rendered

**Comment by kariabancroft** [comment]
Thank you for pre-emptively deleting the db files :-)

File: app/controllers/admin/categories_controller.rb [hunk 3]
```diff
@@ -25,8 +25,7 @@ def destroy
 
   def new_or_edit
     @categories = Category.find(:all)
-    @category = Category.new
-    @category = Category.find(params[:id]) if params[:id]
+    params[:id] ? @category = Category.find(params[:id]) : @category = Category.new
```
Comment (rmtolmach) [inline app/controllers/admin/categories_controller.rb]: Does this look better?

Comment (kariabancroft) [inline app/controllers/admin/categories_controller.rb]: Yep! I think you can simplify it even more by putting the `@category =` at the start of the line

Comment (rmtolmach) [inline app/controllers/admin/categories_controller.rb]: nice! thanks

**Comment by kariabancroft** [comment]
Ok! Good to close!

**Pull request closed without merging**